### PR TITLE
RCPS 203 204 205: EDIT STUDENTS all fields, contact number fields, birthday fields

### PR DESCRIPTION
**Pull request by StevenRabacal**
RCPS 203: stud-info-mgt-app:EDITSTUDENT- contact number field
Defect Description: I had input 12 digits on the contact number field and the system accepts it.
Enter 12 digits
Expected Result: display “ Invalid input“. and not accept numbers beyond 11 digits
Actual Result: The system accepts wrong input and does not display “Invalid input“.

RCPS 204: stud-info-mgt-app:EDITSTUDENT- Birthday field
Defect Description: I had input beyond 2024 birthday, and the system accepts it.
Expected Result: display “ Invalid input“. and not accept numbers beyond 2024
Actual Result: The system accepts wrong input and does not display “Invalid input“.

RCPS 205: stud-info-mgt-app:EDITSTUDENT- ALL FIELDS
Defect Description: I had input non-alpha numeric characters and system still accepts it.
Enter non alpha numeric characters
Expected Result: display “ Invalid input“. and not accept the input with non-alphanumeric
Actual Result: The system accepts wrong input and does not display “Invalid input“.

### CODE REVIEW

**Comment by StevenRabacal** [comment]
RCPS 203: stud-info-mgt-app:EDITSTUDENT- contact number field

RCPS 204: stud-info-mgt-app:EDITSTUDENT- Birthday field

RCPS 205: stud-info-mgt-app:EDITSTUDENT- ALL FIELDS

File: src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java [hunk 1]
```diff
@@ -161,9 +175,21 @@ protected void onUpdateStudOClicked(ActionEvent event){
...
+        return input != null && input.matches("[a-zA-Z0-9]+");
+    }
+
     private boolean isValidEmail(String email) {
         String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,}$";
         return email != null && email.matches(emailRegex);
     }
-
+    private boolean isValidContactNumber(String contactNumber) {
+        return contactNumber.matches("\\d{1,11}");
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java]: ensure that the system strictly accepts 11 numbers only not either 1 to 11

File: src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java [hunk 2]
```diff
@@ -104,6 +104,10 @@ private void showDashboard() {
 
     @FXML
     protected void onUpdateStudOClicked(ActionEvent event){
+        if (!isValidInput()) {
+            System.err.println("Invalid input: Non-alphanumeric characters are not allowed.");
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java]: use Alert

File: src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java [hunk 3]
```diff
@@ -120,6 +124,11 @@ protected void onUpdateStudOClicked(ActionEvent event){
             System.err.println("Invalid Email Address: " + enteredEmail);
             return;
         }
+        String contactNumber = contactNo.getText().trim();
+        if (!isValidContactNumber(contactNumber)) {
+            System.err.println("Invalid Contact Number: " + contactNumber);
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java]: use Alert

File: src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java [hunk 4]
```diff
@@ -128,6 +137,11 @@ protected void onUpdateStudOClicked(ActionEvent event){
         LocalDate selectedDate = birthday.getValue();
         if (selectedDate != null) {
             try {
+                LocalDate minDate = LocalDate.now().minusYears(17);
+                if (selectedDate.isAfter(minDate)) {
+                    System.err.println("Invalid date: Please select a date from 17 years ago or earlier.");
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/UpdateStudentController.java]: use Alert for errors

**Review by zhairanicolepacheco — CHANGES_REQUESTED** [review]
see comments

**Review by zhairanicolepacheco — APPROVED** [review]

**Pull request merged**